python/tk_hiero_export/sg_shot_processor.py

```diff
@@ -246,35 +300,283 @@ def startProcessing(self, exportItems):
+                "timecode_cut_item_out": tc_cut_item_out,
+                "timecode_edit_in": tc_edit_in,
+                "timecode_edit_out": tc_edit_out,
+            })
+
+            # add the populated cut item data to the list for batching later
+            cut_item_data_list.append(cut_item_data)
+
+            if cut_order == 1:
+                # first item in the cut, set the cut's start timecode
+                cut_data["timecode_start"] = tc_edit_in
+
+            if cut_order == len(cut_related_tasks):
+                # last item in the cut, set the cut's end timecode
+                cut_data["timecode_end"] = tc_edit_out
+
+        # all tasks processed, add the duration to the cut data
+        cut_data["duration"] = cut_duration
+
+        # create the cut to get the id.
+        sg = self.app.shotgun
+        cut = sg.create("Cut", cut_data)
+        self._app.log_debug("Created Cut in Shotgun: %s" % (cut,))
+        self._app.log_info("Created Cut '%s' in Shotgun!" % (cut["code"],))
+
+        # build a list of batch requests for the cut items
+        batch_data = []
+        for cut_item_data in cut_item_data_list:
+
+            # make sure the cut items have a parent Cut
+            cut_item_data["cut"] = {"id": cut["id"], "type": "Cut"}
+
+            batch_data.append({
+                "request_type": "create",
+                "entity_type": "CutItem",
+                "data": cut_item_data,
+            })
+
+        # batch create the cut items
+        self.app.log_info("Creating CutItems in Shotgun...")
+        cut_items = sg.batch(batch_data)
+        self._app.log_info("Finished creating CutItems in Shotgun!")
+        self._app.log_debug("Created CutItems in Shotgun: %s" % (cut_items,))
+
+        # attach the newly created cut items to their corresponding tasks.
+        # the transcode task will update the cut item with the version
+        # after it is created.
+        for cut_item in cut_items:
+            cut_order = cut_item["cut_order"]
+
+            # because these cut related tasks are in order, we can get the
+            # corresponding tasks for this cut item.
+            (shot_updater_task, transcode_task) = cut_related_tasks[cut_order - 1]
+
+            # dont' want to assume that there is an associated transcode task.
+            # if there is, attach the cut item data so that the version is
+            # updated. If not, then we'll get a cut item without an associated
+            # version (cut info only in SG, nothing playable).
+            if transcode_task:
+                transcode_task._cut_item_data = cut_item
+
+    def _timecode(self, frame, fps):
+        """Convenience wrapper to convert a given frame and fps to a timecode.
+
+        :param frame: Frame number
+        :param fps: Frames per seconds (float)
+        :return: timecode string
+        """
+        return hiero.core.Timecode.timeToString(frame, fps,
+            hiero.core.Timecode.kDisplayTimecode)
```

Comment:
Hey @sgsteph, since we don't have an official timecode format or field and these are just text fields I think that making sure that what you see in Hiero is what you see in Shotgun is the right move.

Definitely a different story if Hiero didn't already have a good timecode API to leverage and we were duplicating code, but that's not the case here.